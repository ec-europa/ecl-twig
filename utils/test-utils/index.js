const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
