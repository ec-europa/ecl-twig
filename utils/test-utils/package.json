{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
