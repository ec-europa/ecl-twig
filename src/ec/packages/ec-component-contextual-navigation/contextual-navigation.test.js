import path from 'path';
import { renderTwigFileAsNode } from '@ecl-twig/test-utils';

import demoData from './demo/data';

describe('EC - Contextual Navigation', () => {
  demoData.itemMore.icon.path = 'static/icons';
  const template = path.resolve(__dirname, './contextual-navigation.html.twig');
  const render = params => renderTwigFileAsNode(template, params);

  test('renders correctly', () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });
});
