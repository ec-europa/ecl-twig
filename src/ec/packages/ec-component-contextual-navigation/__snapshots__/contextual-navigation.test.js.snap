// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Contextual Navigation renders correctly 1`] = `
<nav
  class="ecl-contextual-navigation"
  data-ecl-contextual-navigation=""
>
  <div
    class="ecl-contextual-navigation__label"
  >
    Label for contextual navigation
  </div>
  <ul
    class="ecl-contextual-navigation__list"
    data-ecl-contextual-navigation-list=""
  >
    <li
      class="ecl-contextual-navigation__item"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-contextual-navigation__link"
        href="/example"
      >
        Item 1
      </a>
    </li>
    <li
      class="ecl-contextual-navigation__item"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-contextual-navigation__link"
        href="/example"
      >
        Item 2
      </a>
    </li>
    <li
      class="ecl-contextual-navigation__item"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-contextual-navigation__link"
        href="/example"
      >
        Item 3
      </a>
    </li>
    <li
      class="ecl-contextual-navigation__item ecl-contextual-navigation__item--collapsed"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-contextual-navigation__link"
        href="/example"
      >
        Item 4
      </a>
    </li>
    <li
      class="ecl-contextual-navigation__item ecl-contextual-navigation__item--collapsed"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-contextual-navigation__link"
        href="/example"
      >
        Item 5
      </a>
    </li>
    <li
      class="ecl-contextual-navigation__item ecl-contextual-navigation__item--collapsed"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-contextual-navigation__link"
        href="/example"
      >
        Item 6
      </a>
    </li>
    <li
      class="ecl-contextual-navigation__item  ecl-contextual-navigation__item--more"
    >
      <button
        class="ecl-button ecl-button--ghost ecl-contextual-navigation__more"
        data-ecl-contextual-navigation-more="true"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            More
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--fluid ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            data-ecl-contextual-navigation-more="true"
            focusable="false"
          >
            <use
              xlink:href="static/icons#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </li>
  </ul>
</nav>
`;
